### ISSUE TITLE
Knowledge transfer with A2C transformer actor-model

### ISSUE THREAD

**Issue by resialre** [MolecularAI/SMILES-RL#1]
Thank you for your work.

Have you considered how implementing a transformer as the actor model may impact the generative capabilities of your transformer? I have tried doing so, but due to computational limitations, I am not so sure of the results. Still, I was able to generate some interesting results. 

Using knowledge transfer on the pretrained RNN policy, I was able to pretrain a decoder-only transformer to predict some novel scaffolds and compounds. However, this was within 500 training steps and a batch size of 32. 

Here is the link to the repository (I am a novice coder): 
[https://github.com/resialre/SMILES-RL-Transformer.git](url)

Here is a preliminary run comparing the A2C with the actor model as either the original RNN with LSTM cell-type or the pre-trained transformer run on an actor learning rate of 1e-6:
![image](https://github.com/MolecularAI/SMILES-RL/assets/2024129/873f5985-9d53-444b-9e74-5cf3106a5eb8)

**Comment by hampusgs** [comment]
Thank you for showing interest in this work!

I have considered trying a transformer architecture instead, but have not had the time to implement it. Some recent work by [1] observed no major improvement in using transformers, but I am sure that there is a possible gain in using transformers instead. 

By looking at your preliminary run, there seems to be something off. I do not know what score you are using, but I would expect the RNN-based A2C to learn more in 500 steps. It seems like it has not learned anything since the scores are more or less constant. This could of course be due to your low batch size, but I would expect a monotonic improvement. 

You are welcome to share any future results. 

[1] https://arxiv.org/pdf/2310.02902

**Comment by hampusgs** [comment]
Thank you for showing interest in this work!

I have considered trying a transformer architecture instead, but have not had the time to implement it. Some recent work by [1] observed no major improvement in using transformers, but I am sure that there is a possible gain in using transformers instead.

By looking at your preliminary run, there seems to be something off. I do not know what score you are using, but I would expect the RNN-based A2C to learn more in 500 steps. It seems like it has not learned anything since the scores are more or less constant. This could of course be due to your low batch size, but I would expect a monotonic improvement.

You are welcome to share any future results.

[1] https://arxiv.org/pdf/2310.02902
